cruise-control-core/src/main/java/com/linkedin/cruisecontrol/httframeworkhandler/CruiseControlRequestContext.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2021 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+package com.linkedin.cruisecontrol.httframeworkhandler;
+
+import io.vertx.core.MultiMap;
+import java.io.IOException;
+import java.util.Map;
+
+public interface CruiseControlRequestContext<T> {
+
+    String[] HEADERS_TO_TRY = {
+            "X-Forwarded-For",
+            "Proxy-Client-IP",
+            "WL-Proxy-Client-IP",
+            "HTTP_X_FORWARDED_FOR",
+            "HTTP_X_FORWARDED",
+            "HTTP_X_CLUSTER_CLIENT_IP",
+            "HTTP_CLIENT_IP",
+            "HTTP_FORWARDED_FOR",
+            "HTTP_FORWARDED",
+            "HTTP_VIA",
+            "REMOTE_ADDR"
+    };
+
+    String getRequestURL();
+
+    String getUserTaskIdString();
+
+    String getMethod();
+
+    String getPathInfo();
+
+    String getClientIdentity();
+
+    MultiMap getHeaders();
+
+    String getHeader(String header);
+
+    String getRemoteAddr();
+
+    Map<String, String[]> getParameterMap();
+
+    String getRequestUri();
+
+    String getParameter(String parameter);
+
+    void writeResponseToOutputStream(int responseCode,
+                                     boolean json,
+                                     boolean wantJsonSchema,
+                                     String responseMessage,
+                                     T config) throws IOException;
```

Comment:
We could remove this parameter and instead pass it in through the constructors of the subclasses. That way we won't need the generic and this weird handling of this parameter.

Reply:
I have a fix for this, will share with you soon.